internal/fs/dir_handle.go

```diff
@@ -218,15 +229,30 @@ func readAllEntries(
 	return
 }
 
+// localFileEntries lists the local files present in the directory.
+// Local means that the file is not yet present on GCS.
+func localFileEntries(d inode.DirInode, localFileInodes map[inode.Name]inode.Inode) (localEntries []fuseutil.Dirent, err error) {
```

Comment:
Thinking loud,  when we implement that rmDir and renameDir doesnt support when localfiles are present, we might need this logic in fs.go. See if you need this logic and if yes move this to fs.go

Reply:
I initially added this method to DirInode, but realised that all the methods in DirInode today deals with GCS objects. That will be the most appropriate location from code reusability POV as well. Do you think it will be okay to move this to DirInode?

Reply:
Yeah should be good, if that operations have dirInode with them.